### PR TITLE
Patterned optogenetic series without pattern table

### PR DESCRIPTION
**Pull request by alessandratrapani**
This PR is for a version of the extension that does not contain pattern table. Link directly one optogenetic pattern to the series

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
TODO

- [x] update UML to be up to date with the rollback on this PR

File: spec/ndx-holographic-stimulation.extensions.yaml [hunk 1]
```diff
@@ -1,8 +1,132 @@
...
+  doc: An extension of Device to include the Light Sorce metadata
+  attributes:
+  - name: stimulation_wavelength
+    dtype: float32
+    doc: stimulation wavelength in nm
+  - name: filter_description
+    dtype: text
+    doc: filter_description
+    required: false
+  - name: power
```
Comment (CodyCBakerPhD) [inline spec/ndx-holographic-stimulation.extensions.yaml]: Should this be `peak_power`? Or `max_power`? Should it be on the `LightSource` device at all, or a property of the pattern or only the series data itself?

Trying to decouple as much potentially conflicting fields as possible; the Pinto lab for example had a device of `700 W` max power but they tuned it down to `400` for some stimulations

**Comment by CodyCBakerPhD** [comment]
@bendichter @pauladkisson @weiglszonja 

Aside from the known issue with the default `device` display theoretically fixed on dev HDMF, and aside from the data modelling concerns of #11, I'd say this version of the extension is good for official review by the rest of the group as an early version of this extension that could get approved and merged to serve as a base for further perfection in follow-ups

The metadata in particular looks quite nice aside from my question about where `power` belongs

I've added you all to @weiglszonja's embargoed dandiset on staging, which is also being shared with Jeremy to help develop better Neurosift visualizations on this type of data; the file `sub-NCCR32_ses-Into-the-void-t-series-stim-000_behavior+ophys.nwb` has an example of this extension applied to real experiment data

File: spec/ndx-holographic-stimulation.extensions.yaml (hunk 1 above)
Comment (alessandratrapani) [reply spec/ndx-holographic-stimulation.extensions.yaml]: I will leave peak_power in the LightSource as a property of the pulse laser (not required), and express the power of the stimulus in the data

**Review by pauladkisson — CHANGES_REQUESTED** [review]
Is there a reason to reference the stimulation device in two places? It seems like we could avoid some potential errors and extra complexity by only referencing the stimulation device in one object (PatternedOptogeneticSeries).

Everything else looks pretty nice to me.

**Comment by alessandratrapani** [comment]
> Is there a reason to reference the stimulation device in two places? It seems like we could avoid some potential errors and extra complexity by only referencing the stimulation device in one object (PatternedOptogeneticSeries).

I don't think we can remove it from PatternedOptogeneticSite, since device is a property of the original OptogeneticSite from which it extends. But I will keep it in PatternedOptogeneticSeries because it makes more sense I think. 
(is the same thing as in TwoPhotonSeries and ImagingPlane). 
I already took into consideration this if we are going to include the ogen module in the microscopy NEP.

**Pull request merged**